File: HFSGuardaDiretorio_GtkSharp_C#/gui/FrmPrincipalProgresso.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 08/07/2015
 * Time: 11:33
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using Gtk;
using HFSGuardaDiretorio.comum;

namespace HFSGuardaDiretorio.gui
{
	/// <summary>
	/// Description of FrmPrincipalProgresso.
	/// </summary>
	public class FrmPrincipalProgresso : IProgressoLog
	{
	    FrmPrincipal form;
	
	    public FrmPrincipalProgresso(FrmPrincipal form) 
	    {
	        this.form = form;
		}
	    
		public void ProgressoLog(Progresso progresso)
		{
			/*
	        if (progresso.Posicao == 0) {
				this.form.PBar.Minimum = progresso.Minimo;
				this.form.PBar.Maximum = progresso.Maximo;
				this.form.PBar.PulseStep = progresso.Passo;
			}
			*/
			this.form.PBar.Fraction = Rotinas.calculaProgresso(progresso.Maximo, progresso.Posicao);				
		}
	    
	}
}

File: HFSGuardaDiretorio_SharpDevelop_Python/gui/FrmPrincipalProgresso.py

from comum import IProgressoLog

class FrmPrincipalProgresso(IProgressoLog.IProgressoLog):
	def __init__(self, form):
		self._form = form

	def ProgressoLog(self, progresso):
		if progresso.Posicao == 0:
			self._form.pb.Minimum = progresso.Minimo
			self._form.pb.Maximum = progresso.Maximo
			self._form.pb.Step = progresso.Passo
		self._form.pb.Value = progresso.Posicao